agent/05_agent_create_cluster.sh

```diff
@@ -102,9 +102,60 @@ function wait_for_cluster_ready() {
   echo "Cluster is ready!"
 }
 
+function wait_for_operator() {
+    subscription="$1"
+    namespace="${2:-}"
+    echo "Waiting for operator ${subscription} to get installed on namespace ${namespace}..."
+
+    for _ in $(seq 1 60); do
+        csv=$(oc -n "${namespace}" get subscription "${subscription}" -o jsonpath='{.status.installedCSV}' || true)
+        if [[ -n "${csv}" ]]; then
+            if [[ "$(oc -n "${namespace}" get csv "${csv}" -o jsonpath='{.status.phase}')" == "Succeeded" ]]; then
+                echo "ClusterServiceVersion (${csv}) is ready"
+                return 0
+            fi
+        fi
+
+        sleep 10
+    done
+
+    echo "Timed out waiting for csv to become ready!"
+    return 1
+}
+
+function create_lso_operator_v4_11() {
+  if [[ ${OPENSHIFT_RELEASE_STREAM} == "4.12" || ${OPENSHIFT_RELEASE_STREAM} == "4.13" ]]; then
+    oc delete subscription local-storage-operator -n openshift-local-storage
+    cat > "${mceManifests}/lso_operator_4_11.yaml" << EOF
+apiVersion: operators.coreos.com/v1
+kind: OperatorGroup
+metadata:
+ name: local-operator-group
+ namespace: openshift-local-storage
+spec:
+ targetNamespaces:
+   - openshift-local-storage
+---
+apiVersion: operators.coreos.com/v1alpha1
+kind: Subscription
+metadata:
+ name: local-storage-operator
+ namespace: openshift-local-storage
+spec:
+ installPlanApproval: Automatic
+ name: local-storage-operator
+ source: redhat-operators-v4-11
+ sourceNamespace: openshift-marketplace
+EOF
+    apply_manifest "$mceManifests/lso_operator_4_11.yaml"
+    wait_for_operator "local-storage-operator" "openshift-local-storage"
+  fi
+}
+
 function mce_prepare_postinstallation_manifests() {
   local mceManifests=$1
-
+  # TODO: Remove this once LSO is published to the 4.12 catalog.
+  create_lso_operator_v4_11
```

Comment:
If we think this same issue can happen in the future, e.g. when 4.13 is being released and its operator is not released yet we should make this generic, for example, call it "create_lp_operator_latest_available' and use a variable to contain the latest - 4.11 in this case

Reply:
Changes have been implemented!